### DISCUSSION TITLE
Enable Playwright video with Serenity JS

### DISCUSSION THREAD

**Discussion by vishalmgada** [serenity-js/serenity-js#1670] [How to... ?] [answered]
Hi Could you please share the configuration required for enabling video recording for playwright tests with Serenity JS?

We are facing challenges in using actor and video recording feature of playwright.

**Comment by jan-molak** [comment, marked as answer]
Hi @vishalmgada, video recording is a [Playwright feature](https://playwright.dev/docs/videos) that needs to be enabled in the Playwright configuration.

## Serenity/JS with Playwright Test

If you're using Serenity/JS with Playwright Test, you will do that in the `playwright.config.ts` file (see [Serenity/JS + Playwright Test project template config](https://github.com/serenity-js/serenity-js-playwright-test-template/blob/main/playwright.config.ts) for reference). This file configures Playwright browser context, which Serenity/JS then uses to instantiate the ability to `BrowseTheWebWithPlaywright` provided to your actors:

```typescript
import { devices } from '@playwright/test';
import type { PlaywrightTestConfig } from '@serenity-js/playwright-test';

/**
 * See https://playwright.dev/docs/test-configuration.
 */
const config: PlaywrightTestConfig = 

  use: {
    video: 'on-first-retry',
  },

  // other configuration
}
```

## Serenity/JS with Cucumber and Playwright

If you're using Serenity/JS with Cucumber and Playwright, you need to instantiate the ability to `BrowseTheWebWithPlaywright` yourself (see [Serenity/JS + Cucumber + Playwright project template config](https://github.com/serenity-js/serenity-js-cucumber-playwright-template/blob/dd1af55ede39e0f34a1a0d16483469a0a54a5b8b/features/support/serenity.config.ts#L39-L43) for reference). 

When instantiating the ability, you can additionally provide [Playwright options](https://serenity-js.org/api/playwright/interface/PlaywrightOptions), including the option to [`recordVideo`](https://serenity-js.org/api/playwright/interface/PlaywrightOptions#recordVideo):

```typescript
    configure({

        // Configure Serenity/JS actors to use Playwright browser
        actors: new Actors(browser, {
            baseURL:     'https://serenity-js.org/',
            recordVideo: { dir: 'videos/' }
        }),
    });
```
